operator/app/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -403,6 +402,60 @@ private void configureTLS(Deployment deployment) {
             deployment.getSpec().getTemplate().getSpec().getVolumes().add(volume);
             kcContainer.getVolumeMounts().add(volumeMount);
         }
+
+        var userRelativePath = readConfigurationValue(Constants.KEYCLOAK_HTTP_RELATIVE_PATH_KEY);
+        var kcRelativePath = (userRelativePath == null) ? "/" : userRelativePath;
+        var protocol = (this.keycloakCR.getSpec().isHttp()) ? "http" : "https";
+        var kcPort = (this.keycloakCR.getSpec().isHttp()) ? Constants.KEYCLOAK_HTTP_PORT : Constants.KEYCLOAK_HTTPS_PORT;
+
+        var baseProbe = new ArrayList(List.of("curl", "--head", "--fail", "--silent"));
+
+        if (!this.keycloakCR.getSpec().isHttp()) {
+            baseProbe.add("--insecure");
+        }
+
+        var readyProbe = new ArrayList<>(baseProbe);
+        readyProbe.add(protocol + "://127.0.0.1:" + kcPort + kcRelativePath + "/health/ready");
+        var liveProbe = new ArrayList<>(baseProbe);
+        liveProbe.add(protocol + "://127.0.0.1:" + kcPort + kcRelativePath + "/health/live");
+
+        kcContainer
+                .getReadinessProbe()
+                .setExec(new ExecActionBuilder().withCommand(readyProbe).build());
+        kcContainer
+                .getLivenessProbe()
+                .setExec(new ExecActionBuilder().withCommand(liveProbe).build());
+    }
+
+    public String readConfigurationValue(String key) {
+        if (this.keycloakCR != null &&
+                this.keycloakCR.getSpec() != null &&
+                this.keycloakCR.getSpec().getServerConfiguration() != null
+        ) {
+            var serverConfigValue = this.keycloakCR
+                    .getSpec()
+                    .getServerConfiguration()
+                    .stream()
+                    .filter(sc -> sc.getName().equals(key))
+                    .findFirst();
+            if (serverConfigValue.isPresent()) {
+                if (serverConfigValue.get().getValue() != null) {
+                    return serverConfigValue.get().getValue();
+                } else {
+                    var secretSelector = serverConfigValue.get().getSecret();
+                    var secret = client.secrets().inNamespace(getNamespace()).withName(secretSelector.getName()).get();
```

Comment:
I believe we can get an NPE here if `secretSelector == null`. I know we should theoretically in any case get an error from here:
https://github.com/keycloak/keycloak/blob/ea22989d8963a6089a0e357fbf4bb409de3800c7/operator/app/src/main/java/org/keycloak/operator/v2alpha1/WatchedSecretsStore.java#L174
But still for debugging purposes, I think we could throw some more meaningful exception. 🙂 WDYT?